sdk/maps/maps-search/README.md

```diff
@@ -0,0 +1,166 @@
+
+```bash
+npm install @azure/identity
+```
+
+You will also need to register a new AAD application and grant access to Azure Maps by assigning the suitable role to your service principal. Please refer to the [Manage authentication](https://docs.microsoft.com/azure/azure-maps/how-to-manage-authentication) page.
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables: `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.
+
+You will also need to specify the Azure Maps resource you intend to use by specifying the `clientId` in the client options.
+The Azure Maps resource client id can be found in the Authentication sections in the Azure Maps resource. Please refer to the [documentation](https://docs.microsoft.com/azure/azure-maps/how-to-manage-authentication#view-authentication-details) on how to find it.
+
+```javascript
+const { MapsSearchClient } = require("@azure/maps-search");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const credential = new DefaultAzureCredential();
+const client = new MapsSearchClient(credential, "<maps-account-client-id>");
+```
+
+#### Using a Subscription Key Credential
+
+You can authenticate with your Azure Maps Subscription Key.
+
+```javascript
+const { MapsSearchClient, AzureKeyCredential } = require("@azure/maps-search");
+
+const credential = new AzureKeyCredential("<subscription-key>");
+const client = new MapsSearchClient(credential);
+```
+
+## Key concepts
+
+### MapsSearchClient
+
+`MapsSearchClient` is the primary interface for developers using the Azure Maps Search client library. Explore the methods on this client object to understand the different features of the Azure Maps Search service that you can access.
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common Azure Maps Search tasks, including:
+
+- [Request latitude and longitude coordinates for an address](#request-latitude-and-longitude-coordinates-for-an-address)
+- [Search for an address or Point of Interest](#search-for-an-address-or-point-of-interest)
+- [Make a Reverse Address Search to translate coordinate location to street address](#make-a-reverse-address-search-to-translate-coordinate-location-to-street-address)
+- [Translate coordinate location into a human understandable cross street](#translate-coordinate-location-into-a-human-understandable-cross-street)
+
+### Request latitude and longitude coordinates for an address
+
+You can use an authenticated client to convert an address into latitude and longitude coordinates. This process is also called geocoding. In addition to returning the coordinates, the response will also return detailed address properties such as street, postal code, municipality, and country/region information.
+
+```javascript
+const searchResult = await client.searchAddress("400 Broad, Seattle");
```

Comment:
We should also add example about how to manipulate the `searchResult` (e.g. use `console.log(xxx)` to show how to get members/resutls from `searchResult`). Same for other examples

Reply:
Updated in eb39dbb6202458a04fb2ed04d7315085e04f0694